### DISCUSSION TITLE
How to trigger action for tag on old commit

### DISCUSSION THREAD

**Discussion by dmfrodrigues** [community/community#25963] [Actions]
<p>So I have a repository, and I want to tag an old version of it (i.e. not the most recent commit).</p>
<p>I have a workflow that is activated when a tag is pushed:</p>
<pre><code class="lang-auto">on:
  push:
    tags:
      - 'v*'
</code></pre>
<p>I tried creating a new tag in GitHub for a recent version that not the most recent version, and it did not trigger this action. I also tried using <code>git tag v1.0.0 &lt;not-the-most-recent-commit&gt;</code> and pushed it, and it also didn’t get triggered.</p>
<p>Why is that and how can I trigger the action no matter if the tagged commit is the most recent or not?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/dmfrodrigues">@dmfrodrigues</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>This is by designed. When you tag old commit and push to repository, github will check if event trigger exists on the workflow yaml “AT THE COMMIT”, while not the latest workflow yaml file.</p>
<p>You can checkout the files based on <code>the old commit</code>, if there’s already tag trigger(on: push: tags) on the workflow yaml, tag on this commit, the workflow will be triggered. If no tag trigger, workflow will not be triggered.</p>
<p>My <a href="https://github.com/weide-zhou/ticket30/runs/864332002?check_suite_focus=true" rel="nofollow noopener">workflow</a> for your reference.</p>
<p>It’s not supported to always trigger actions no matter old/recent commit is tagged.</p>
<p>Hope it’s clear and helpful!</p>

**Comment by dmfrodrigues** [comment]
<p>I see. So the workflow is only triggered if the commit being tagged has the workflow yml file for tags, right?</p>
<p>Please just confirm I understood it right, so I can set it as solution.</p>
<p>Thanks.</p>

**Comment by weide-zhou** [comment]
<p>Yes, it’s correct!<a class="mention" href="/dmfrodrigues">@dmfrodrigues</a></p>
<p>The github.ref is the updated ref, if you tag on old commit and push to repository, it will check the yaml file till the commit, trigger the workflow if the tag trigger exists.</p>
<p>Thanks.</p>
